Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp

```diff
@@ -1010,5 +1041,56 @@ namespace EMotionFX
         {
             return HasEditorMaterialComponent() ? AZ::Edit::PropertyVisibility::Hide : AZ::Edit::PropertyVisibility::Show;
         }
+
+        void EditorActorComponent::DetachFromEntity()
+        {
+            if (!m_actorInstance)
+            {
+                return;
+            }
+
+            ActorInstance* attachedTo = m_actorInstance->GetAttachedTo();
+            if (attachedTo)
+            {
+                attachedTo->RemoveAttachment(m_actorInstance.get());
+            }
+            AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetParent, AZ::EntityId());
+        }
+
+        void EditorActorComponent::OnActorInstanceCreated(ActorInstance* actorInstance)
+        {
+            auto it = AZStd::find(m_attachments.begin(), m_attachments.end(), actorInstance->GetEntityId());
```

Comment:
nit: consider naming this `attachmentIt` to make it a bit clear what it is referring to

Reply:
ok i will change this one.